Xenakios/TakeRenaming.cpp

```diff
@@ -223,40 +165,39 @@ void DoRenameTakeAndSourceFileDialog(COMMAND_T* ct)
 					if (newfnsplit[2].compare(fnsplit[2])!=0)
 						newfilename.append(fnsplit[2]);
 
-					MoveFile(oldname.c_str(),newfilename.c_str());
+					if (!MoveFile(oldname.c_str(), newfilename.c_str()))
```

Comment:
This does not prompt if `newfilename` already exists on non-Windows, where `MoveFile` is an alias for `rename` (defined in swell-types.h:75).

This condition would only be entered if the move fails (eg. permission issue, directory not found...). Offlining first is only useful on Windows.

Reply:
> This does not prompt if newfilename already exists on non-Windows, where MoveFile is an alias for rename (defined in swell-types.h:75).
Using `FileOrDirExists()` from `SnM_Util.h` instead should be ok?

> Offlining first is only useful on Windows.
Tbh not sure what this implies. :)
For `PCM_source::SetFileName()` (which is called later on) it says in `reaper_plugin.h`:
> this will only be called when offline
so it needs to be set offline anyway?

Reply:
> Using FileOrDirExists() from SnM_Util.h instead should be ok?

Yes.

> PCM_source::SetFileName() [...] this will only be called when offline

Ah, I didn't know that. Then yes it does need to be set offline first.